can/constructor-hydrate/constructor-hydrate.js

```diff
@@ -0,0 +1,93 @@
+/**
+ * @module {connect.Behavior} can-connect/can/constructor-hydrate/constructor-hydrate
+ * @parent can-connect.behaviors
+ *
+ * Persists instances with the same id.
+ *
+ * @signature `constructorHydrate( baseConnection )`
+ *
+ * 	 Overrides [can-define/map/map]'s `setup` method and checks whether a newly created instance already
+ * 	 exists in [can-connect/constructor/store/store.instanceStore]. If it exists that instance will be
+ * 	 returned instead of a new object.
+ *
+ * 	 This behavior has to be used with [can-connect/constructor/store] and [can-connect/can/map/map] behaviors.
+ *
+ * @body
+ *
+ * ## Use
+ *
+ * This behavior could be useful if `Type` converters of [can-define/map/map] are used in multiple places of your app.
+ * In which case if a property is set with an id of an already created instance then the connection behaviour will
+ * check [can-connect/constructor/store]. If there is already an instance with the same id then it will be returned
+ * instead of a new object.
+ *
+ * Let's say we have the following page state with two properties which are of type `Student`:
+ * ```javascript
+ * 	const myPage = new (DefineMap.extend({
+ * 	    student: { Type: Student },
+ * 	    teamLead: { Type: Student },
+ * 	}));
+ * ```
+ *
+ * The type `Student` is a DefineMap with `can-connect` capabilities:
+ * ```javascript
+ * 	const Student = DefineMap.extend({});
+ * 	Student.List = DefineList.extend({
+ * 	    '#': { Type: Student }
+ * 	});
+ *
+ * 	const StudentConnection = connect([
```

Comment:
connections aren't constructor functions.  Also, with merge-map and such, we should show everyone:

```
Student.connection = connect( ... 
```